### ISSUE TITLE
MVP - make it work

### ISSUE THREAD

**Issue by messense** [messense/multi-lsp-proxy#1]
- [ ] Merge server capabilities
- [ ] Forward requests based on the capabilities of child LSPs
- [ ] Merge responses if a request is sent to multiple child LSPs? Not sure.

I'd like to move to using `tower-lsp`, waiting for https://github.com/ebkalderon/tower-lsp/issues/284 to settle.

**Comment by techee** [comment]
FYI, I made it work but by creating a different project and writing it in Python ;-)

https://github.com/techee/lsp-proxy

I implemented this as I preferred this solution for my [geany-lsp](https://github.com/techee/geany-lsp) plugin to having this functionality implemented by the plugin itself.

I don't know if you plan improving your project but if you do, apart from the pointes mentioned above you also have to synchronize `initialize` and `shutdown` requests and send only a single response to the client once all the servers return the response. I've written more about what has to be done in the lsp-proxy's [README](https://github.com/techee/lsp-proxy?tab=readme-ov-file#lsp-proxy).

I'm interested in any feedback if anyone wants to give the proxy a try - I only tested it with Geany where it works but there might be some problems with other editors.
